Add CI to repo

laurenchilutti opened the pull request:
**Description**

Adding CI tests to this repo.  This brings over a similar workflow as is done in the GFDL_atmos_cubed_sphere repository.  Will build solo model (nh, sw, and hydro) and then run the 35 tests in RTS/CI.

GitHub recognizes yaml files stored in `.github/workflows` as GitHub Actions workflows:
`Intel_Parallelworks_CI.yaml` specifies instructions for automated CI with every pull request.  There are 4 steps to the CI:
-*checkout*: Executes a script `checkout.sh` stored on Parallelworks cloud.  This script will checkout the SHiELD_build directory (checks out the branch/commit of the PR that triggers the CI)  and check out component codes.  We only compile FMS when the CHECKOUT_code script specifies a new version of FMS.  We store the latest version of FMS for use in compiling SHiELD on Parallelworks at /contrib/fv3/2023.2.0/SHiELD_build/externallibs.  
-*build*:  Executes the script `compile.sh` with various build configurations (a total of 36 combinations, of which any with sw + shield will be skipped).  The script executes a compile command inside of the CI container on Parallelworks
-*test*: Runs the script `run_test.sh` 35 times.  Each time giving it an argument specifying which test to run.  These tests run in parallel.

The scripts stored on parallelworks will be version controlled in this repository and are at `.github/.parallelworks`.  

Fixes # (issue)

**How Has This Been Tested?**

Tested by the successful checks on this PR

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

laurenchilutti committed:
Create Intel_Parallelworks_CI.yaml

laurenchilutti committed:
Create daily_cleanup_parallelworks.yaml

laurenchilutti committed:
Update Intel_Parallelworks_CI.yaml

kaiyuan-cheng committed:
Update CHECKOUT_code

laurenchilutti committed:
Update Intel_Parallelworks_CI.yaml

laurenchilutti committed:
Update daily_cleanup_parallelworks.yaml

laurenchilutti committed:
Update Intel_Parallelworks_CI.yaml

laurenchilutti committed:
Update Intel_Parallelworks_CI.yaml

laurenchilutti committed:
Add Parallelworks CI scripts to repo

laurenchilutti commented:
I have one more update to make.  I will send an email when I am ready for this to be reviewed for a final time.

bensonr reviewed (commented):
I notice some scripts are invoking bash and others are invoking sh, is there a reason you can't use one shell throughout?

If the argument list may grow, you may want to consider arg-parsing for the future https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash.

laurenchilutti committed:
Create Intel_Parallelworks_multbuild.yaml

laurenchilutti committed:
update path to external libs

laurenchilutti committed:
compile with a matrix of options and cleanup arument parsing

laurenchilutti committed:
fix typos in compile.sh

laurenchilutti committed:
fixing arg parsing

laurenchilutti committed:
update Intel_Parallelworks_CI.yaml to allow for up to 17 concurrent compiles.

laurenchilutti committed:
updating all parallelworks CI files to be bash.

laurenchilutti commented:
@bensonr I can't see any reason why I had some as sh and some as bash.  They should all be consistent now.  I also modified the argument parsing to be cleaner.  

This is now ready for review.  Thanks!

bensonr reviewed (approved):
This looks good.  Thanks for considering the suggested actions.

laurenchilutti committed:
Merge remote-tracking branch 'noaa/main' into AddCI

laurenchilutti committed:
Merge branch 'main' into AddCI

laurenchilutti committed:
Refactor Intel Parallelworks CI workflow

Updated the CI workflow to simplify build and test configurations.

laurenchilutti committed:
Enhance CHECKOUT_code for 'all' configuration support

Updated configuration options to include 'all' and modified related logic for handling input arguments. Added support for checking out all configurations in the script.

laurenchilutti committed:
Update Intel_Parallelworks_CI.yaml

laurenchilutti committed:
removed buildfms workflow

lharris4 reviewed (approved):
Looks good to me. Is there any list of the scripts run by `run_test.sh`?
